### PR TITLE
Codeship CI/CD for python client

### PR DESCRIPTION
**Pull request by goatrocks**
This uses enables CI/CD for the python client using versioneer and codeship. 

~~Currently this draft PR pushes to test.pypi.org, with two test pushes here done using jet locally: https://test.pypi.org/project/indico-client/#history~~

Updated to have creds for pypi.org.

Here's a shot of the tests running in codeship:
image

The tests will run on any push right now, but can be configured to just run on PR.

The deploy step will only run on creation of a new tag following the x.y.z versioning. Pypi expects pep440 version strings, so pre-release clients can be released by creating a pre-release tag like x.y.za0 or x.y.zrc1. The version will be automatically adjusted in setup.py and when the code calls get_version() in _version.py.

edit: Now with tox for parallel testing of multiple python versions

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+#put token in user's home dir
+printenv INDICO_API_TOKEN > ~/indico_api_token.txt
+#install additional test reqs and run smoke tests
+pip install requests-mock pytest
```
Comment (pastap) [inline scripts/test.sh:5]: what's the reasoning to not use the tests_requires with setup.py in the Dockerfile?

File: codeship-steps.yml [hunk 2]
```diff
@@ -1,8 +1,7 @@
-- type: parallel
-  steps:
-    - name: Python 2 Tests
-      service: python2
-      command: nosetests -sv .
-    - name: Python 3 Tests
-      service: python3
-      command: nosetests -sv .
+- name: test
+  service: indico-client
```
Comment (pastap) [inline codeship-steps.yml:2]: can we try and retain our parallel python tests to check 2 and 3? or are we deprecating python 2 support?

File: scripts/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+rm -rf /indico-client/dist
+pip install twine
+python setup.py sdist
+echo $TWINE_USERNAME
+echo $TWINE_REPOSITORY
+twine upload --repository testpypi /indico-client/dist/*
```
Comment (pastap) [inline scripts/deploy.sh:7]: assuming you'll sub in $TWINE_REPOSITORY here? or is there a reason not to hardcode it?

File: scripts/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+rm -rf /indico-client/dist
+pip install twine
```
Comment (pastap) [inline scripts/deploy.sh:3]: should this be ver-locked?

File: codeship-steps.yml (hunk 2 above)
Comment (goatrocks) [reply codeship-steps.yml:2]: That is a good question, we absolutely could have a parallel run of the tests using python 2 or multiple versions of python, but are we planning to continue to support python 2 given the sunset date has passed over a year ago?

File: scripts/deploy.sh (hunk 3 above)
Comment (goatrocks) [reply scripts/deploy.sh:7]: I'll sub in $TWINE_RESPOSITORY or (more likely) remove it entirely, for now I just wanted to be really sure I didn't push anything anywhere besides testpypi.

File: scripts/test.sh (hunk 1 above)
Comment (goatrocks) [reply scripts/test.sh:5]: I wasn't sure if that would stay around when setup.py test was deprecated

File: scripts/deploy.sh (hunk 4 above)
Comment (goatrocks) [reply scripts/deploy.sh:3]: Most likely! I'll add that

File: codeship-steps.yml (hunk 2 above)
Comment (goatrocks) [reply codeship-steps.yml:2]: Okay, per slack discussion it sounds like it's safe to drop python 2 support, but I did add in tox for parallel testing of multiple versions of 3, although I'll want to polish that some more

File: tox.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:20.04
+
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -y apt-transport-https
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y software-properties-common && add-apt-repository ppa:deadsnakes/ppa
+
+# ensure the base image has what we need
+#RUN DEBIAN_FRONTEND=noninteractive apt-get -y install build-essential software-properties-common
```
Comment (jkgenser) [inline tox.Dockerfile:9]: Do you need this anymore or want to keep for reference?

Comment (goatrocks) [reply tox.Dockerfile:9]: Hmm, no, don't need it any more as far as I can tell. I'll remove it so I don't leave commented out stuff!

**Pull request merged**